tests/test_syntax/extensions/test_abbr.py

```diff
@@ -383,6 +446,47 @@ def test_abbr_with_attr_list(self):
             extensions=['abbr', 'attr_list']
         )
 
+    def test_abbr_superset_vs_subset(self):
+        self.assertMarkdownRenders(
+            self.dedent(
+                """
+                abbr, SS, and abbr-SS should have different definitions.
+
+                *[abbr]: Abbreviation Definition
+                *[abbr-SS]: Abbreviation Superset Definition
+                *[SS]: Superset Definition
+                """
+            ),
+            self.dedent(
+                """
+                <p><abbr title="Abbreviation Definition">abbr</abbr>, """
+                + """<abbr title="Superset Definition">SS</abbr>, """
+                + """and <abbr title="Abbreviation Superset Definition">abbr-SS</abbr> """
+                + """should have different definitions.</p>
+                """
+            )
+        )
+
+    def test_abbr_empty(self):
+        self.assertMarkdownRenders(
+            self.dedent(
+                """
+                *[abbr]: Abbreviation Definition
+
+                abbr
+
+                *[abbr]:
+
+                Testing document text.
+                """
+            ),
+            self.dedent(
+                """
+                <p>abbr</p>\n<p>Testing document text.</p>
+                """
+            )
+        )
```

Comment:
I don't think this is the correct behavior for this test case. The syntax for definitions are modeled after reference style links. Notice how an empty link reference behaves:

```markdown
[empty link reference][empty]

[empty]: 
```

The above renders as

```html
<p>[empty link reference][empty]</p>
<p>[empty]: </p>
```

Specifically, an empty link reference is not recognized as one at all. I think this is probably how abbreviation definitions should behave as well. The fact that a test for an empty definition didn't exist suggests that this edge case wasn't given any consideration in the past. Thanks for catching another bug.

What this means is that, given the current input, the expected output for `test_abbr_empty()` would be:

```html
<p><abbr title="Abbreviation Definition">abbr</abbr></p>
<p>*[abbr]:</p>
<p>Testing document text.</p>
```

Of course, that would also mean that your attempt to create a way to override/remove a previously defined abbreviation would no longer work. Perhaps a reasonable workaround would be to add support for wrapping the definition in quotes. Then empty quotes would be parsed as an empty definition.

```markdown
*[not-a-def]:
*[empty-def]: ""
```

Reply:
On the other hand, there is no mechanism to predefine link references. However, with this change we now have a way to predefine abbr definitions. Therefore, it is reasonable to expect to be able to define  an empty definition to override/remove it. Therefore, maybe we can just leave things as-is. I'll need to think some more about this.

Reply:
I like the more declarative nature of the `*[empty-def]: ""` syntax.
Potential alternatives to `""`: `[]`, `{}`, `_`, `...`

There are likely others and I don't know which ones feel more Markdown-ish.

Reply:
Quotes are used in both inline and reference style [links](https://daringfireball.net/projects/markdown/syntax#link) for titles.  In reference style links the title can also be wrapped in regular parentheses (`()`), but in all my years I have never seen anyone actually use that. I suppose maybe it could be useful if the content needed to include quote characters. However, here we only need the wrapping characters when the content is empty, so that would be less of a concern. I think using quotes (supporting both single and double) makes the most sense in this context.

Reply:
Added handling for ignoring `*[not-a-def]:` and removing definitions with `*[empty-def]: ""` or `*[empty-def]: ''`
Also, while testing I found that `*[]: Bad Definition` resulted in abbr tags everywhere, so added testing to avoid that.